docs/DeveloperGuide.adoc

```diff
@@ -153,94 +153,234 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+
+.Import Command Sequence Diagram
+image::ImportCommandSequenceDiagram.png[]
+
+The arguments passed to the Import Command will be parsed by the `ImportCommandParser` class. +
+If the given arguments are valid, a new `ImportCommand` object will be returned.
+In this class, invalid arguments will result in a `ParseException`. +
+
+Two types of checks will be done before returning the `ImportCommand`:
+
+* Check whether the file is exists in the data folder.
+* Check whether the first prefix is `orderType` prefix.
 
 [NOTE]
-If the `currentStatePointer` is at index `addressBookStateList.size() - 1`, pointing to the latest address book state, then there are no undone address book states to restore. The `redo` command uses `Model#canRedoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
+====
+`CsvProcessor` is  a helper class that helps to retrieve the data from the csv file and process the data before
+giving to `ImportCommand`.
+====
+
+After the `ImportCommand` object is being returned to the `LogicManager`, the `LogicManager` will start
+to run the execute the import Command, this is will be shown at the <<Execution of Import Command Sequence Diagram, Execution of Import Command Sequence Diagram>>.
 
-Step 5. The user then decides to execute the command `list`. Commands that do not modify the address book, such as `list`, will usually not call `Model#commitAddressBook()`, `Model#undoAddressBook()` or `Model#redoAddressBook()`. Thus, the `addressBookStateList` remains unchanged.
+==== Execution of Import Command Sequence Diagram
 
-image::UndoRedoState4.png[]
+.Import Command Sequence Diagram 2
+image::ImportCommandSequenceDiagram2.svg[]
 
-Step 6. The user executes `clear`, which calls `Model#commitAddressBook()`. Since the `currentStatePointer` is not pointing at the end of the `addressBookStateList`, all address book states after the `currentStatePointer` will be purged. We designed it this way because it no longer makes sense to redo the `add n/David ...` command. This is the behavior that most modern desktop applications follow.
+The `execute(model)` function of the `ImportCommand` will first check if the data given starts with
+*order* or *return* and pass to the `InsertCommandParser` or `ReturnCommandParser` respectively. +
 
-image::UndoRedoState5.png[]
+Afterwards, `InsertCommandParser` or `ReturnCommandParser` will return the `InsertCommand` or `ReturnCommand` respective
+if it successfully parse the data. The `ImportCommand` will then call the `execute(Model)` function of `InsertCommand`
+or `ReturnCommand`. This will cause a delivery order or return order being added into the `Model`.
 
-The following activity diagram summarizes what happens when a user executes a new command:
+The `ImportCommand` will call its own `printResult()` function and return a String message to the `CommandResult` object
+which is then pass back to the `LogicManager`.
+**
```

Comment:
Would this be a minor grammatical error? Would the ** be part of the  formatting?